### DISCUSSION TITLE
What do the major version bumps mean in the project?

### DISCUSSION THREAD

**Discussion by kurko** [ObsidianPublisher/obsidian-github-publisher#328] [General]
I have a contribution to make (soon, hopefully), but now I noticed another major version bump, from v6 to v7. I see the project using the Major.Minor.Patch format, so I'm assumed it's semver, but I'm not sure if the major bump really is a breaking change.

I thought it was a major breaking change, but then I saw the [CHANGELOG](https://github.com/ObsidianPublisher/obsidian-github-publisher/blob/master/CHANGELOG.md#700-2024-02-26) and it's merely a bug fix.

The intent with the question is to understand how stable the APIs are, otherwise it's hard to work on contributions if the code keeps changing. I mean, it's all good if it is a breaking change, progress is good. It's just to organize my own time on how to best contribute (and to avoid rework on my end).

**Comment by Mara-Li** [comment]
Hello!
The last major was because the new version only works with the last update of Obsidian, because I use a new API!

**Reply by Mara-Li** [reply]
Here the link to the 7.0 initial version : https://github.com/ObsidianPublisher/obsidian-github-publisher/releases/tag/7.0.0

Also sorry for my spaghetti code :')
